packages/table/components/CheckboxTable.tsx

```diff
@@ -7,24 +7,52 @@ export interface CheckboxTableProps extends TableProps {
   /**
    * Callback for when a user makes a selection. Passes selected row data as a parameter
    */
-  onChange?: (selectedRows: object[]) => void;
+  onChange?: (selectedRows: {}) => void;
+  /**
+   * The unique key in each row's data object.
+   *
+   * Ideally, the value of this key in your table's data
+   * should be a number or string
+   */
+  uniqueKey: string;
+}
+
+export interface CheckboxTableState {
+  headerChecked: boolean;
 }
 
-class CheckboxTable extends React.PureComponent<CheckboxTableProps> {
+class CheckboxTable extends React.PureComponent<
+  CheckboxTableProps,
+  CheckboxTableState
+> {
   constructor(props) {
     super(props);
 
+    this.state = {
+      headerChecked: false
+    };
+
     this.getSelectedRows = this.getSelectedRows.bind(this);
     this.checkboxCellRenderer = this.checkboxCellRenderer.bind(this);
     this.getHeaderCheckbox = this.getHeaderCheckbox.bind(this);
   }
 
+  public componentWillReceiveProps(nextProps) {
+    const selectedLength = Object.keys(nextProps.selectedRows).length;
+    if (Boolean(selectedLength) && selectedLength === this.props.data.length) {
+      this.setState({ headerChecked: true });
```

Comment:
Originally, I wasn't using state. I was using a mutable variable in `getHeaderCheckbox`. Dima requested I change it to state in his review.

If I just use `checked={this.state.headerChecked}`, it will not be checked on mount if a user passes all rows to `selectedRows`.